uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -196,13 +196,21 @@ define([
     }
 
     /**
-     * Converts given date to server format
+     * Converts given date to server format.
+     * At first, it attempts to parse the date from the timepicker format,
+     * if it fails, then it falls back to format auto-detection.
```

Comment:
why would it fail? (just a question)

Reply:
Tests were passing, but they threw a warning about fall-back-to-native-date-object deprecation and that bothered me. So I investigated and noticed that date is passed in several formats:

- date object
- moment object
- string UI input (DD/MM/YYYY) - was not supported

So, failing means: It tries to parse by `DD/MM/YYYY` and if it fails, it falls back to the format auto-detection.

Now no tests in CiviHR throw this warning and pass meaning we handle date formats for parsing correctly.